codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -233,13 +241,34 @@ private static FlowState state(Http2Stream stream) {
         private int lowerBound;
         private boolean endOfStream;
 
-        FlowState(Http2Stream stream, int initialWindowSize) {
+        /**
+         * The associated {@link #stream} has transitioned into a state which requires the full amount of
+         * state for flow control.
+         */
+        void streamActivated() {
+            // TODO (scottmitch) should this object act as a proxy object to another object that holds the state?
```

Comment:
I'm not sure we would want to null anything out until the stream is actually removed anyway.  Maybe I'm missing your point?